src/nvim/ex_getln.c

```diff
@@ -2263,20 +2280,56 @@ static void draw_cmdline(int start, int len)
     msg_outtrans_len(ccline.cmdbuff + start, len);
 }
 
+void ui_ext_cmdline_char(int c, int shift)
```

Comment:
If this pattern is followed it's easy to undo later. The converse is probably not.

In most of these cases the function is already full of a lot of noise. Giving the reader a signal "go here for the externalized logic" is helpful. The fact that some cases the logic is small just means when a few lines are added here and there later, we have to have this discussion again.